test/helpers.ts

```diff
@@ -31,3 +44,42 @@ export function createFakeEditor(path?: string): TextEditor {
   editor.getBuffer().setPath(path || "/a/b/c/d.js")
   return editor
 }
+
+/** A function that sleeps for a preiod of `ms` and then resolves the given value */
+function sleep<T>(resolveValue: T, ms: number) {
+  return new Promise<T>((resolve) => {
+    setTimeout(() => resolve(resolveValue), ms)
+  })
+}
+
+export function createFakeLanguageServerProcess(): LanguageServerProcess {
+  spawn("lsp", [], { stdio: "inherit", shell: true })
+    .exit(() => sleep(0, 500))
```

Comment:
It doesn't look like `exit` actually awaits this promise.
https://github.com/wraithgar/spawk/blob/main/lib/interceptor.js#L115

```suggestion
    .exit(0)
```

Reply:
No, it does
https://github.com/wraithgar/spawk/blob/1baa72ad31d8cf3e439c0882f91bcfe8d3ba44f6/lib/interceptor.js#L211